src/memory/AmdFlash.cc

```diff
@@ -436,9 +440,9 @@ uint16_t AmdFlash::peekCFI(size_t address) const
 	}
 }
 
-bool AmdFlash::isSectorWritable(size_t sector) const
+bool AmdFlash::isWritable(const Sector& sector) const
 {
-	return vppWpPinLow && (sector == one_of(0u, 1u)) ? false : (writeAddress[sector] != -1) ;
+	return vppWpPinLow && (sector == one_of(sectors[0], sectors[1])) ? false : !sector.writeProtect;
```

Comment:
This may go wrong when there is only a single sector (thus `sectors[1]` is out of bounds). But I guess that's not possible. And I also see that this code is still re-written in later patches.

Reply:
Yeah, indeed never occurs in practice, and consider it a temporary “hack” just to keep it working, that’s addressed in the next patch.